packages/editor/src/components/block-list/block.js

```diff
@@ -91,7 +91,7 @@ export class BlockListBlock extends Component {
 		}
 
 		if ( this.props.isSelected && ! prevProps.isSelected ) {
-			this.focusTabbable();
+			this.focusTabbable( true );
```

Comment:
What is it we're trying to do here? And why is it inconsistent with the behavior in `componentDidMount`?

Reply:
Okay, so if it's to avoid "select" actually causing one of its inner blocks to become selected, can't this also happen in a newly-mounted block as well?

Reply:
Maybe we should reconsider whether to call `focusTabbable` on a selection update as well. I don't recall what we need it for.

Reply:
> Maybe we should reconsider whether to call focusTabbable on a selection update as well. I don't recall what we need it for.

Initial implementation here was to avoid selecting innerBlocks in all cases but this breaks the writing flow when inserting columns (it actually broke 2 e2e tests) because the first paragraph was not focused in that case. I did consider whether it's fine to not focus the first paragraph in that case as assume we hit "tab" or click no the appender of the column block first but I think from a "Writing flow" perspective, it's better if we focus the first paragraph in a column directly when creating a columns block.

Reply:
>it's better if we focus the first paragraph in a column directly when creating a columns block.

Personally I've found this behavior to be a weird / unexpected side effect of focus behaviors. On the Layout<->Writing spectrum, Columns falls more on the former, so I don't necessarily see the paragraph creation / focus being a necessity.

Reply:
> As "Columns" evolve, it'd be natural to focus on the container when it is added rather than on the paragraph within.

Yes one idea I had was to rewrite the columns block similarly to the table block where you select the number of columns before typing but I though this was out of the scope of this PR anyway.

Reply:
I'm getting a sense we're becoming very ad hoc with how and when we expect focus transitions to occur upon block selection, hacking in various conditions to preserve intended flows while losing sight of what those intended flows are. It's signalling to me that naively checking for / comparing on `isSelected` from `BlockListBlock` is not serving us very well, and that perhaps it ought to be moved / refactored to more appropriately capture the user interactions where we expect it to occur: as best I can tell, this is mostly in splitting / merging / replacing actions occurring while typing within a `RichText`.